mne/preprocessing/tests/test_ica.py

```diff
@@ -1070,4 +1071,35 @@ def test_ica_eeg():
                 ica.get_sources(inst)
 
 
+@testing.requires_testing_data
+def test_read_ica_eeglab():
+    """Test read_ica_eeglab function."""
+    fname = op.join(test_base_dir, "EEGLAB", "test_raw.set")
+    fname_cleaned_matlab = op.join(test_base_dir, "EEGLAB",
+                                   "test_raw.cleaned.set")
+
+    raw = read_raw_eeglab(fname, preload=True)
+    raw_eeg = _check_load_mat(fname, None)
+    raw_cleaned_matlab = read_raw_eeglab(fname_cleaned_matlab,
+                                         preload=True)
+
+    mark_to_remove = ["manual"]
+    comp_info = raw_eeg.marks["comp_info"]
+
+    if len(comp_info["flags"].shape) > 1:
+        ind_comp_to_drop = [np.where(flags)[0]
+                            for flags, label in zip(comp_info["flags"],
+                                                    comp_info["label"])
+                            if label in mark_to_remove]
+        ind_comp_to_drop = np.unique(np.concatenate(ind_comp_to_drop))
+    else:
+        ind_comp_to_drop = np.where(comp_info["flags"])[0]
+
+    ica = read_ica_eeglab(fname)
+    raw_cleaned = ica.apply(raw.copy(), exclude=ind_comp_to_drop)
```

Comment:
@christian-oreilly are the bad ica components marked in the set files? if so, why not reading it and setting on the read the ica.exclude attribute?

Reply:
These marks are from an extension to EEGLAB named [vised-marks](https://github.com/BUCANL/Vised-Marks) which allows to mark epochs, channels and components. These are marked with a label that is defined by the user (e.g., manual, eog, ecg, ...). Which components should be removed depends on the intended usage and the user decision. In this specific case, when I created the test file in Matlab, I marked three channels using the string "manual" and saved in EEGLAB a version of the file where these components were rejected to simulate an artifact rejection preprocessing. Thus, for comparison in the test, I do the same thing in MNE. I don't think that it can be generalized so that components are marked in ica.exclude when importing the ICA object.